faiss/impl/ResidualQuantizer.cpp

```diff
@@ -354,9 +377,16 @@ void ResidualQuantizer::compute_codes(
             residuals.data(),
             distances.data());
 
+    std::vector<float> norms;
+    if (search_type == ST_norm_float || search_type == ST_norm_qint8 ||
+        search_type == ST_norm_qint4) {
+        norms.resize(n);
+        fvec_norms_L2sqr(norms.data(), x, d, n);
```

Comment:
This may lead to a high quantization error of vectors although the norm is accurate. A more general way is to store the norm of reconstructed vectors.

Reply:
I think that storing the original norms gives better accuracy. Will redo experiments to confirm.

Reply:
OK you're right, encoding the compressed norms is more accurate.